File: 백준/Silver/10815. 숫자 카드/숫자 카드.py
import sys

N = int(sys.stdin.readline().strip())
numsN = set(map(int, sys.stdin.readline().split()))
M = int(sys.stdin.readline().strip())
numsM = list(map(int, sys.stdin.readline().split()))

result = []
for numM in numsM:
    if numM in numsN:
        result.append(1)
    else:
        result.append(0)

print(" ".join(map(str, result)))

File: 백준/Silver/10815. 숫자 카드/숫자 카드.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int N = Integer.parseInt(br.readLine().trim());
        StringTokenizer st = new StringTokenizer(br.readLine());
        HashSet<Integer> numsN = new HashSet<>(N);
        for (int i = 0; i < N; i++) {
            numsN.add(Integer.parseInt(st.nextToken()));
        }
        int M = Integer.parseInt(br.readLine().trim());
        st = new StringTokenizer(br.readLine());
        br.close();
        int[] numsM = new int[M];
        for (int i = 0; i < M; i++) {
            numsM[i] = Integer.parseInt(st.nextToken());
        }

        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < M; i++) {
            if (numsN.contains(numsM[i])) {
                sb.append("1 ");
            } else {
                sb.append("0 ");
            }
        }
        
        bw.write(sb.toString().trim() + "\n");
        bw.flush();
        bw.close();
    }
}